custom_model_runner/datarobot_drum/drum/model_metadata.py

```diff
@@ -151,12 +163,20 @@ def read_model_metadata_yaml(code_dir) -> PythonTypingOptional[dict]:
                 if "typeSchema" in model_config:
                     revalidate_typeschema(model_config["typeSchema"])
                 model_config = model_config.data
-            except YAMLError as e:
+            except YAMLValidationError as e:
                 if "found a blank string" in e.problem:
                     print("The model_metadata.yaml file appears to be empty.")
                 else:
                     print(e)
                 raise SystemExit(1)
+            except StrictYAMLError as e:
+                raise DrumFormatSchemaException(
+                    "\nStrictYAMLError: The current format does not comply with strict yaml rules."
+                    " (Empty list on fields are not allowed)\n{}".format(e)
+                )
```

Comment:
Why for some exceptions we print and exit but for this one we raise a new exception class? From the looks of it, these all appear to be similar classes of errors (e.g. bad user input) so I'm surprised we are behaving differently in this way. Should all of them be raising an exception -- is DRUM server swallowing the errors we print?

Reply:
Yes the prints error are not propagated to logs shows in the DataRobot App, when some of these error happen the logs display in the UI only show `Error drum: 1`.

We could consolidate all exceptions into one common that is a good idea, but I guess we should raise exceptions for all error related to user input.